File: python/book_suggestion_system/book_suggestion_system.py
import random

books = ["The Mystery", "Student Life", 
        "Poor Dad, Rich dad", "Growth Mindset",
        "The Hobbit", "Animal Farm", "Brave kingdom" ]


def get_suggestion():
    random_number = random.randint(1,101)
    random_book = random.choice(books)
    
    return f"Book Title: {random_book}\nPage: {random_number}"
    
def add_book(book):

    if book in books:
    
        return "Book already exists."
    else:
        books.append(book)   
    
    return "Book added successfully."


def remove_book(book_name):

    if book_name in books:
    
        books.remove(book_name)
        
        return "Book removed succefully!."
    else:
        return "Book does not exist."


def update_book(old_book, new_book):

    if old_book in books:

        index = books.index(old_book)

        books[index] = new_book
        
        return "Book updated succefully!"
    else:
        return "Book does not exist."



def show_book():
    
    all_books = ""
    
    count = 1
        
    for book in books:

        all_books += f"{count}. {book}\n"
        
        count += 1
        
    return all_books

File: java/BookSuggestionSystem/BookSuggestionSystem.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Random;


public class BookSuggestionSystem{

    public static ArrayList<String> bookStore( ){
    
        String [] books = {"The Mystery", 
                           "Student Life", 
                           "Poor Dad, Rich dad", 
                           "Growth Mindset",
                           "The Hobbit", 
                           "Animal Farm", 
                           "Brave kingdom", 
                           "Peaceful Life"};
 
            ArrayList<String> allBookList = new ArrayList<>(Arrays.asList(books)); 

        return allBookList;
    
    }

    public static String suggestBook(ArrayList<String> bookList){

        Random randomCollector = new Random();
        
        int randomNumber = randomCollector.nextInt(1, 100);
        
        int randomIndex = randomCollector.nextInt(bookList.size());
            String bookSuggetion = bookList.get(randomIndex);
        
        return String.format("Book Title: %s %nPasge: %d%n", bookSuggetion, randomNumber );
        
        } 
        
        
    public static String addBook(ArrayList<String> bookStore, String bookTitle){
                        
            if (bookStore.contains(bookTitle)){

                return "Book already exist";
                }
            else{

                    bookStore.add(bookTitle);                    
                return "Book added succefully";
            }
        } 
        
    public static String removeBook(ArrayList<String> bookStore, String bookTitle){
    
            if (!bookStore.contains(bookTitle)){

                return "Book does not exist";
                }
            else{

                    bookStore.remove(bookTitle);                    
                return "Book removed succefully";
            }
        } 
        
        
    public static String updateBook(ArrayList<String> bookStore, String oldBookTitle, String newBookTitle){
    
            if (bookStore.contains(oldBookTitle)){
                int index = bookStore.indexOf(oldBookTitle);
                    bookStore.set(index, newBookTitle);
        
        return "Book updated succefully!";
        }
    else{
        return "Book does not exist.";                         
        }
        
  } 
  
  
    public static String showBooks(ArrayList<String> bookStore){
    
            String allBooks = "";
            int count = 1;
            for(String book : bookStore){
            
                allBooks += String.format("%d. %s\n", count, book);
                count++;
            }
              return allBooks;
  
  }


}

